src/Primitives/test/StringValuesTests.cs

```diff
@@ -221,6 +221,33 @@ public void ImplicitStringConverter_Works()
             Assert.Equal<string[]>(new string[] { aString }, stringValues);
         }
 
+        [Fact]
+        public void GetHashCode_SingleValueVsArrayWithOneItem_SameHashCode()
+        {
+            var sv1 = new StringValues("value");
+            var sv2 = new StringValues(new[] { "value" });
+            Assert.Equal(sv1, sv2);
+            Assert.Equal(sv1.GetHashCode(), sv2.GetHashCode());
+        }
+
+        [Fact]
+        public void GetHashCode_NullVsArrayWithNullItem_DifferentHashCodes()
+        {
+            var sv1 = new StringValues((string)null);
+            var sv2 = new StringValues(new[] { (string)null });
+            Assert.NotEqual(sv1, sv2);
+            Assert.Equal(sv1.GetHashCode(), sv2.GetHashCode());
```

Comment:
Do we expect the two hash codes to be equal in this use case?

Reply:
No? Maybe factor in the array length?

Reply:
@Tratcher why not have them be equal? The idea here is to treat a single item array as the same as the single item itself.

Reply:
`(string)null` and `(string[])null` are considered equal default null values, but `new[] { (string)null }` is not, you had to actually specify that value.

Reply:
https://github.com/aspnet/Extensions/blob/c89d414a99d735ae8e6f8b5c46ea4669b38dd09d/src/Primitives/src/StringValues.cs#L436-L454

Reply:
It doesn't matter if the hash codes are the same if Equals still fails.

Reply:
Do we want Equals and GetHashCode to have a coherent behavior? If so, then would Equals need to change to include GetHashCode in its calculation?

Reply:
I found the MemberData `FilledStringValuesWithExpected` which is used in a test called `Equals_StringArray`, showing different cases where two StringValues can be equal. I am guessing whatever the final implementation of GetHashCode will be has to make sure it returns the same hashcode for those instances that Equals returns true for:

```
       public static TheoryData<StringValues, string[]> FilledStringValuesWithExpected
        {
            get
            {
                return new TheoryData<StringValues, string[]>
                {
                    { default(StringValues), new string[0] },
                    { StringValues.Empty, new string[0] },
                    { new StringValues(string.Empty), new[] { string.Empty } },
                    { new StringValues("abc"), new[] { "abc" } },
                    { new StringValues(new[] { "abc" }), new[] { "abc" } },
                    { new StringValues(new[] { "abc", "bcd" }), new[] { "abc", "bcd" } },
                    { new StringValues(new[] { "abc", "bcd", "foo" }), new[] { "abc", "bcd", "foo" } },
                    { string.Empty, new[] { string.Empty } },
                    { "abc", new[] { "abc" } },
                    { new[] { "abc" }, new[] { "abc" } },
                    { new[] { "abc", "bcd" }, new[] { "abc", "bcd" } },
                    { new[] { "abc", "bcd", "foo" }, new[] { "abc", "bcd", "foo" } },
                    { new[] { null, "abc", "bcd", "foo" }, new[] { null, "abc", "bcd", "foo" } },
                    { new[] { "abc", null, "bcd", "foo" }, new[] { "abc", null, "bcd", "foo" } },
                    { new[] { "abc", "bcd", "foo", null }, new[] { "abc", "bcd", "foo", null } },
                    { new[] { string.Empty, "abc", "bcd", "foo" }, new[] { string.Empty, "abc", "bcd", "foo" } },
                    { new[] { "abc", string.Empty, "bcd", "foo" }, new[] { "abc", string.Empty, "bcd", "foo" } },
                    { new[] { "abc", "bcd", "foo", string.Empty }, new[] { "abc", "bcd", "foo", string.Empty } }
                };
            }
        }
```

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.object.gethashcode?view=netframework-4.8#remarks
"Two objects that are equal return hash codes that are equal. However, the reverse is not true: equal hash codes do not imply object equality, because different (unequal) objects can have identical hash codes."